### DISCUSSION TITLE
blog/2023/10/17/syncing-my-kindle-with-calibre-web-koreader-and-syncthing/

### DISCUSSION THREAD

**Discussion by giscus** [extrange/personal-blog#6] [Announcements]
# blog/2023/10/17/syncing-my-kindle-with-calibre-web-koreader-and-syncthing/

None

https://nicholaslyz.com/blog/2023/10/17/syncing-my-kindle-with-calibre-web-koreader-and-syncthing/

**Comment by nuovodna** [comment]
Thanks for your detailed post! I'm in a similar situation but a little bit different. I have a Calibre library with calibre-server enabled and i read books through my web browser and with my Koreader (on a kobo device). I'd like to synch my koreader highlights to my calibre library, how can i achieve this? Thanks!

**Reply by extrange** [reply]
Hey there, thanks for reading! Unfortunately I haven't tried highlight syncing - however, if the highlights are stored in a plaintext file somewhere, it might be possible to just use Syncthing to sync that file between your devices.

**Reply by HerdingG** [reply]
Highlights and notes are stored in the same name folder of the e-book you read.

**Comment by kevin-david** [comment]
Thanks for sharing, I just set up something similar with my Kindle since I learned about [WinterBreak](https://kindlemodding.org/jailbreaking/WinterBreak/)

I'm curious if you've noticed an impact on Kindle battery life at all with SyncThing running in the background, presumably all the time?

**Reply by extrange** [reply]
Thanks, I've never heard about WinterBreak. How does it differ from the jailbreak scripts by NiJuLe et al?

So, I've actually stopped using Syncthing. I now turn on Wifi, and download my books from Calibre-Web on the Kindle (Calibre-Web supports an OPDS feed, which KOReader can consume).

**Reply by kevin-david** [reply]
AFAIK it just works with more (newer) devices like my 11th-gen Paperwhite 5.

Interesting about Calibre-Web, SyncThing seems to work decently for me and goes offline when the screen is off so I assume it's getting paused. But haven't been able to ascertain impact on battery life while it's running (yet).

**Reply by extrange** [reply]
I see. I've never liked fiddling around with jailbreaks specific to one device model - so WinterBreak is something I'll try once I get a new device.

**Reply by HerdingG** [reply]
Highlights and notes are stored in the same folder of the e-book you read.

**Comment by HerdingG** [comment]
Thinks for sharing! I think syncthing is a perfect way to sync book with the highlight and note. That is the only way to sync whole floder to sync highlight and note. But I'm curious if this way can synchronize the reading progress?

**Reply by Metrasher** [reply]
I'm no expert. But I don't recommend that based on my experience. It's better to synchronize reading progress using that precise feature itself. That works pretty well.

Metadata synchrony doesn't work well. At least I couldn't manage to accomplish that Syncthing understands that the latest version of the file is the one to go with.
And besides, the metadata file uses absolute paths, so if you use different systems, it won't work, as Kobo has it's path, Kindle uses another, snd so on.

Synchronization of books works perfectly with Syncthing, but I only managed to achieve that and nothing more.
